CMakeLists.txt

```diff
@@ -344,3 +344,6 @@ include(CMake/BuildFlatBuffers.cmake)
 if(FLATBUFFERS_PACKAGE_DEBIAN)
     include(CMake/PackageDebian.cmake)
 endif()
+
+configure_file(flatbuffers.pc.in flatbuffers.pc @ONLY)
```

Comment:
Please add some comments to what this does and why it is needed.

Reply:
I assumed that would be really obvious.

pkg-config is *very* helpful for detecting libraries (and corresponding compiler/linker flags) in a very simple, portable and canonical way. actually it's the standard mechanism since aeons. Actually, I never would like to do that manually again.

Reply:
Sorry, I am no CMake expert, and neither is everyone else.

Reply:
Okay, for a explaination:

Pkg-config is a tool for easily retrieving compiler/linker flags that are needed to link against some library. The idea is: each library package also ships a .pc file that contains the necessary metadata (eg. cflags, ldflags, ...), including dependencies. These can be queried eg. from a makefile (or some genarator like autoconf, etc), so individual applications don't need any special per-library logic/knowledge - just the name and minimal version.

Isolating out those things into a generic tool is also *VERY* helpful for distros that might need some tweaks (eg. processing pathes for crosscompiling -> sysroot, etc), w/o having to patch each individual package.

For example in a makefile you could add things like:
PKG_CONFIG ?= pkg-config
CFLAGS += `$(PKG_CONFIG) --cflags foo`
LIBS += `$(PKG_CONFIG) --libs foo`

The patch for CMakeLists.txt tells cmake to generate the file flatbuffers.pc from template flatbuffers.pc.in, replacing variables w/ the actual build-time values (eg. pathes, etc), and also install that file on the target.

Reply:
Ok, that sounds good.. thanks for explaining.